### DISCUSSION TITLE
Cache serverless functions

### DISCUSSION THREAD

**Discussion by MaybeThisIsRu** [vercel/vercel#4994] [Ideas]
At the moment, serverless functions are built on every single deploy. This adds an astonishing 1.5 minutes to my delpoy time. It would be lovely if these were cached and only rebuilt if the source has changed.

**Comment by styfle** [comment]
Since we don't have restrictions on `import` or `require`, we can't reliably determine if your function+dependencies will change without building the code.

If you're using Next.js with `/pages/api`, there are some optimizations in place.

If you're using `/api` with a different frontend framework, you could try changing the dependencies in `/package.json` to only be for your frontend, and `/api/package.json` to only contain dependencies for your serverless functions.

**Reply by MaybeThisIsRu** [reply]
Will creating a `package.json` for `/api/` improve the build speed? I'd imagine it avoids a full re-build unless anything in the API source changes?

I've moved on and [self-host my API endpoints](https://github.com/hirusi/web-comments) now. Still, I'd be curious to know how a separate package manifest is supposed to work in theory.

**Reply by styfle** [reply]
Each function looks for package.json by starting in its current directory and walking the tree up to the root directory
